sdks/wasm/library_mono.js

```diff
@@ -80,7 +80,19 @@ var MonoSupportLib = {
 		},
 
 		mono_load_runtime_and_bcl: function (vfs_prefix, deploy_prefix, enable_debugging, file_list, loaded_cb, fetch_file_cb) {
-			Module.FS_createPath ("/", vfs_prefix, true, true);
+			// /dev/random doesn't work on js shells, so define our own
+			// See library_fs.js:createDefaultDevices ()
+			var random_device;
+			if (typeof crypto !== 'undefined') {
+				var randomBuffer = new Uint8Array(1);
```

Comment:
This is plain bad code.
We're littering `mono_load_runtime_and_bcl` with something completely unrelated.

mono-random.c should have a `HOST_WASM` version that does invoke into this code, which should be a function by itself.